src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -1887,6 +1887,7 @@ Object addToMarkList(Object guardedObject,
     }
 
     @CoreMethod(names = "set_mark_list_on_object", onSingleton = true, required = 1)
+    @ReportPolymorphism
```

Comment:
This shouldn't have any effect because there is a single specialization instance possible for this node, so no way to trigger multiple and report polymorphism.

Reply:
I confirmed that there is no code generated related to the annotation with this change (I used `graal` master, but I don't think that makes a difference).

Reply:
Did you make `WriteObjectFieldNode` go polymorphic? I didn't see the extra nodes generated until then. I may very well have it wrong, but I saw a bunch of `Integer` splits from the [`argc` checks in cext_ruby](https://github.com/oracle/truffleruby/blob/master/lib/truffle/truffle/cext_ruby.rb#L25-L42) go away after I made this change.

Reply:
> Did you make `WriteObjectFieldNode` go polymorphic?

I didn't try, no, I just looked at whether this annotation changed anything to the generated code and it looked like no, so should have no effect. It's a case Truffle should warn for, I filed an internal issue for that.

The Integer splits sounds strange, maybe something is redefining these Integer methods, even if it installs back the original ones after it will still invalidate the Assumptions permanently.
Could also be some `some_integer == some_nonumeric_object` somewhere, and that would end up calling `some_nonumeric_object == some_integer` which could cause splitting.
But for these Integer#{==,>=} calls they should always use Inlined*Node, unless these methods are redefined at some point, and these inline nodes like InlinedEqualNode do not even use IntegerNode.EqualNode and both sides should be clearly `int` in that code.